src/homework4.py

```diff
@@ -0,0 +1,421 @@
+import sys
+
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from plotly.subplots import make_subplots
+from sklearn.ensemble import RandomForestClassifier
+
+
+def response_type_check(df, response):
+    # Determine if response is continuous or boolean
+    unique_responses = df[response].nunique()
+
+    if unique_responses == 2:
+        boolean_check = True
+        print("Response is boolean\n")
+    else:
+        boolean_check = False
+        print("Response is continuous\n")
+
+    return boolean_check
+
+
+def cat_cont_check(df, predictor):
+    # Determine if the predictor is cat/cont
+    if isinstance(df[predictor][0], float):
+        cat_check = False
+        print("Predictor {} is continuous\n".format(predictor))
+    else:
+        cat_check = True
+        print("Predictor {} is categorical\n".format(predictor))
+
+    return cat_check
+
+
+def bool_response_cat_predictor_plots(df, predictor, response):
+    x = df[predictor]
+    x_type = type(x[0])
+
+    # density_heatmap needs same datatypes and this should work
+    df[response] = df[response].map(x_type)
+
+    plot = px.density_heatmap(
+        df,
+        x=predictor,
+        y=response,
+        title="{} v. {}: Heatmap".format(predictor, response),
+    )
+    plot.show()
+
+
+def bool_response_cont_predictor_plots(df, predictor, response):
```

Comment:
when I ran the titanic dataset with predictor "fare", I got this warning:
Predictor fare is continuous

Optimization terminated successfully.
         Current function value: 0.627143
         Iterations 6
/Users/Downloads/homework4.py:262: RuntimeWarning:

invalid value encountered in long_scalars

Difference with mean of response (unweighted) = nan
Difference with mean of response (weighted) = nan

Not sure what that means really, but the difference with mean of response came back nan.....